pandas/_libs/hashtable_func_helper.pxi.in

```diff
@@ -133,10 +223,11 @@ def duplicated_{{dtype}}({{c_type}}[:] values, object keep='first'):
         {{dtype}}_t value
         {{endif}}
         Py_ssize_t k, i, n = len(values)
-        kh_{{ttype}}_t *table = kh_init_{{ttype}}()
+        kh_{{ttype}}_t *table
         ndarray[uint8_t, ndim=1, cast=True] out = np.empty(n, dtype='bool')
 
-    kh_resize_{{ttype}}(table, min(n, _SIZE_HINT_LIMIT))
+    kh_init(&table)
+    kh_resize(table, <int>min(n, _SIZE_HINT_LIMIT))
```

Comment:
Hmm are you sure this code is equivalent? Something about the sizeof int across various platforms would have me concerned here with the cast

Reply:
Good question.  Without the cast, cython wont compile this.  Maybe we could move the `min` call to its own line, but then we'd need to declare a type for the output and we'd be back where we started.  Unless you're suggesting a different type to cast to?

Reply:
Do you know what the compliation error is? I would at least think we should cast to `Py_ssize_t` since that would match the size of `n` and would hold `_SIZE_HINT_LIMIT`

Reply:
FWIW I think `int` would work for most compilers since `_SIZE_HINT_LIMIT` fits into the 2 ** 21 range, but I think the C standard only requires 16 bits

Niche applications here but I suppose better safe than sorry when casting. Defining `_SIZE_HINT_LIMIT` might also work as it *I think* is just a PyObject atm

Reply:
kh_resize is typed to take an `int`.  I think declaring a type for _SIZE_HINT_LIMIT is worthwhile, but changing <int> casting here less so

Reply:
looks like typing SIZE_HINT_LIMIT breaks non-cython code because it is no longer py-visible.  going to leave this as-is for now

Reply:
Hmm yea that's unfortunate. Can you shared the error the compiler was throwing?

Sorry not trying to be overly difficult on this; I just think casting like this *could* cause very hard to find bugs and am not totally clear on why this would be required now

Reply:
> Sorry not trying to be overly difficult on this; I just think casting like this could cause very hard to find bugs and am not totally clear on why this would be required now

Absolutely no sweat.  This is worth getting right.

> Can you shared the error the compiler was throwing?

So if in hashtable.pyx we change `_SIZE_HINT_LIMIT = (1 << 20) + 7` to `cdef int64_t _SIZE_HINT_LIMIT = (1 << 20) + 7` then we don't get compile-time errors, but do get runtime errors when python modules try to access hashtable._SIZE_HINT_LIMIT because it is not python-visible.

If we _dont_ cdef _SIZE_HINT_LIMIT and don't cast the `min` result, then we get

```
    kh_resize(table, min(n, _SIZE_HINT_LIMIT))
   ^
------------------------------------------------------------

pandas/_libs/hashtable_func_helper.pxi:615:4: Invalid use of fused types, type cannot be specialized
Traceback (most recent call last):
  File "setup.py", line 809, in <module>
    ext_modules=maybe_cythonize(extensions, compiler_directives=directives),
  File "setup.py", line 538, in maybe_cythonize
    return cythonize(extensions, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/Cython/Build/Dependencies.py", line 1096, in cythonize
    cythonize_one(*args)
  File "/usr/local/lib/python3.7/site-packages/Cython/Build/Dependencies.py", line 1219, in cythonize_one
    raise CompileError(None, pyx_file)
Cython.Compiler.Errors.CompileError: pandas/_libs/hashtable.pyx
```